src/tools/wasm-interp.c

```diff
@@ -1203,46 +1253,61 @@ static WasmResult on_register_command(Context* ctx,
 static WasmResult on_assert_unlinkable_command(Context* ctx,
                                                WasmStringSlice filename,
                                                WasmStringSlice text) {
-  char buffer[256];
-  wasm_snprintf(buffer, sizeof(buffer),
-                PRIstringslice ":%d: assert_unlinkable error",
-                WASM_PRINTF_STRING_SLICE_ARG(ctx->source_filename),
-                ctx->command_line_number);
-
-  WasmDefaultErrorHandlerInfo info;
-  info.header = buffer;
-  info.out_file = stdout;
-  info.print_header = WASM_PRINT_ERROR_HEADER_ONCE;
-
-  WasmBinaryErrorHandler error_handler;
-  error_handler.on_error = wasm_default_binary_error_callback;
-  error_handler.user_data = &info;
+  WasmBinaryErrorHandler* error_handler =
+      new_custom_error_handler(ctx, "assert_unlinkable");
 
   ctx->total++;
   char* path = create_module_path(ctx, filename);
+  WasmInterpreterModule* module;
   WasmInterpreterEnvironmentMark mark =
       wasm_mark_interpreter_environment(&ctx->env);
-  WasmResult result = read_module(ctx->allocator, path, &ctx->env,
-                                  &error_handler, &ctx->last_module);
+  WasmResult result =
+      read_module(ctx->allocator, path, &ctx->env, error_handler, &module);
   wasm_reset_interpreter_environment_to_mark(ctx->allocator, &ctx->env, mark);
 
   if (WASM_FAILED(result)) {
     ctx->passed++;
+    result = WASM_OK;
   } else {
     print_command_error(ctx, "expected module to be unlinkable: \"%s\"", path);
-    wasm_free(ctx->allocator, path);
-    return WASM_ERROR;
+    result = WASM_ERROR;
   }
 
   wasm_free(ctx->allocator, path);
-  return WASM_OK;
+  destroy_custom_error_handler(ctx->allocator, error_handler);
+  return result;
 }
 
 static WasmResult on_assert_uninstantiable_command(Context* ctx,
                                                    WasmStringSlice filename,
                                                    WasmStringSlice text) {
-  /* TODO */
-  return WASM_OK;
+  ctx->total++;
+  char* path = create_module_path(ctx, filename);
+  WasmInterpreterModule* module;
+  WasmInterpreterEnvironmentMark mark =
```

Comment:
I don't see a use of local "mark". If the value isn't used, why create a local for it?